### DISCUSSION TITLE
Starting solo-chain network

### DISCUSSION THREAD

**Discussion by weezy20** [paritytech/zombienet#1923] [Q&A]
Is it possible to start solochain networks (substrate-node-template) using zombienet?

**Comment by pepoviola** [comment]
Hi @weezy20, yes is possible. In this case you need to use the config as _relaychain_ for the solochain.

Ping me if you need help.

Thanks!!

**Reply by weezy20** [reply]
I have done the same but the cmd errors with the chainspec json having something missing on some line number when building a raw chainspec out of the plain spec file. I have not encountered this issue doing it manually, but I suspect my chainspec might have some misconfiguration. I'll try to solve that meanwhile and make sure it's not zombienet specific.

**Reply by pepoviola** [reply]
Ahh can you add the error and the chain-spec here to take a look? Thanks!!

**Reply by weezy20** [reply]
Hello again @pepoviola . This time I tried building the `substrate-node` in polkadot-sdk master to ensure there would be no discrepancies. Here's what I found with the following zombinet config toml : 
```toml
[settings]
enable_tracing = false

[relaychain]
# default_image = "{{ZOMBIENET_INTEGRATION_TEST_IMAGE}}"
default_command = "./substrate-node"

chain = "dev"
# chain_spec_path = "chain-spec.json"

  [[relaychain.nodes]]
  name = "alice"
  validator = false
  # db_snapshot="{{DB_SNAPSHOT}}"

  [[relaychain.nodes]]
  name = "bob"
  validator = false
  # db_snapshot="{{DB_SNAPSHOT}}"

  #we need at least 3 nodes for warp sync
  [[relaychain.nodes]]
  name = "charlie"
  validator = false
  # db_snapshot="{{DB_SNAPSHOT}}"

  [[relaychain.nodes]]
  name = "dave"
  validator = false
  # args = ["--sync warp -ldb::blockchain"]
```
Here are the logs from one of the nodes : 
```
It isn't safe to expose RPC publicly without a proxy server that filters available set of RPC methods.
2024-11-23 20:07:39 Substrate Node    
2024-11-23 20:07:39 ✌️  version 3.0.0-dev-08ec8cdbfdb    
2024-11-23 20:07:39 ❤️  by Parity Technologies <admin@parity.io>, 2017-2024    
2024-11-23 20:07:39 📋 Chain specification: Development    
2024-11-23 20:07:39 🏷  Node name: bob    
2024-11-23 20:07:39 👤 Role: FULL    
2024-11-23 20:07:39 💾 Database: RocksDb at /tmp/zombie-4d84c24bc255b8aed85a76408d7a1c69_-110291-6Q1rqGWIpnXi/bob/data/chains/dev/db/full    

====================

Version: 3.0.0-dev-08ec8cdbfdb

   0: sp_panic_handler::set::{{closure}}
   1: std::panicking::rust_panic_with_hook
   2: std::panicking::begin_panic_handler::{{closure}}
   3: std::sys::backtrace::__rust_end_short_backtrace
   4: rust_begin_unwind
   5: core::panicking::panic_fmt
   6: core::option::expect_failed
   7: sc_consensus_grandpa::aux_schema::load_persistent
   8: sc_consensus_grandpa::block_import_with_authority_set_hard_forks
   9: staging_node_cli::service::new_partial
  10: staging_node_cli::service::new_full_base
  11: staging_node_cli::service::new_full
  12: tokio::runtime::context::runtime::enter_runtime
  13: tokio::runtime::runtime::Runtime::block_on
  14: staging_node_cli::command::run
  15: substrate_node::main
  16: std::sys::backtrace::__rust_begin_short_backtrace
  17: std::rt::lang_start::{{closure}}
  18: std::rt::lang_start_internal
  19: main
  20: <unknown>
  21: __libc_start_main
  22: _start


Thread 'main' panicked at 'genesis authorities is non-empty; all weights are non-zero; qed.', /home/dev/pro/polkadot-sdk/substrate/client/consensus/grandpa/src/aux_schema.rs:395

This is a bug. Please report it at:

	https://github.com/paritytech/polkadot-sdk/issues/new

2024-11-23 20:07:45 🔨 Initializing Genesis block/state (state: 0x4241…7d1b, header-hash: 0x1893…3175)    
2024-11-23 20:07:45  creating SingleState txpool Limit { count: 8192, total_bytes: 20971520 }/Limit { count: 819, total_bytes: 2097152 }.    
2024-11-23 20:07:45 👴 Loading GRANDPA authority set from genesis on what appears to be first startup.    
```

**Reply by weezy20** [reply]
And further more, here is the CLI output 
![image](https://github.com/user-attachments/assets/c2668948-6f68-43ff-a497-9b0c9b84eea2)

**Reply by weezy20** [reply]
I think the above errors are being caused due to the tracing setting being enabled.. The following, much simpler config, works like a charm: 
```toml
[relaychain]
default_command = "./substrate-node"
chain = "local"

  [[relaychain.nodes]]
  name = "alice"
  validator = true

  [[relaychain.nodes]]
  name = "bob"
  validator = true
```

**Comment by pepoviola** [comment]
Hi @weezy20, sorry about the delay. I double check this, the issue is that you are not defining any `validator` in the chain, so the authority set result in an empty set. If you set one as `validator` works as expected. Out of curiosity, you need to have `no validators` in your use-case?

Thx!!

**Reply by weezy20** [reply]
So the chain I was trying this out with is a frontier chain using AccountId20 and maybe that's why putting `alice` and `bob` as validator breaks it ?

**Reply by pepoviola** [reply]
Ah, we have the `evm-generic` decorator but works with parachain setups. In you case the solochain is `evm` compatible and must _act_ as relaychain?

**Reply by weezy20** [reply]
Not sure if I understand your question. The solochain is evm compatible yes, but I am defining it as `[relaychain]` nodes because I believe that's the only way to setup a multi-node solo-chain network?

**Reply by pepoviola** [reply]
Yes, that correct. In this case I need to extend the `evm` support to `relaychain` config to make it work. I will create a new issue to track this feature. Can you send me the repo you are using to reproduce on my side?
Thx!

**Reply by weezy20** [reply]
Unfortunately I can't because I do not own the repo, but to replicate it you can start with `polkadot-evm/frontier/template`. We are using that AccountId20 implementation with little to no changes so should work.
